lms/djangoapps/course_api/blocks/urls.py

```diff
@@ -37,4 +37,10 @@
         BlocksInCourseView.as_view(),
         name="blocks_in_course"
     ),
+
+    url(
+        r'^v3/blocks/',
+        BlocksWithCompletionView.as_view(),
+        name="blocks_with_completion_in_course"
+    ),
```

Comment:
this doesn't seem like a breaking change. existing clients won't care that there is a new `completion` field in the response. and it doesn't seem to me like anything in the completion tree-walk would make the performance noticeably slower, right?

so, is there any reason we wouldn't want to keep the API version at `v2`?

Reply:
V2 already has completion in response but it is calculated only on leaf nodes i.e. on unit level. This version propogate completion from parents block till root.

Reply:
is that a breaking change? are there any clients using the V2 API that would break if you made these changes directly to the V2 API? my guess is that they would just happily ignore the new parent-level completion data; would they not?

if we don't need to bump the API version, then it's less work and overhead to leave it at V2.